src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -334,4 +334,18 @@ namespace winrt::TerminalApp::implementation
         }
         args.Handled(true);
     }
+
+    void TerminalPage::_HandleExecuteCommandline(const IInspectable& /*sender*/,
+                                                 const TerminalApp::ActionEventArgs& actionArgs)
+    {
+        if (const auto& realArgs = actionArgs.ActionArgs().try_as<TerminalApp::ExecuteCommandlineArgs>())
+        {
+            _startupActions = TerminalPage::ConvertExecuteCommandlineToActions(realArgs);
+            if (!_startupActions.empty())

```

Comment:
I don't love that we are reusing our startup state for this. Is it possible to split up ProcessStartupActions into `ProcessActions(list_of_actions)`?

Reply:
So, yes-ish. If we wanted to pass the collection of actions as a parameter, there are some options:
* Pass a `std::deque`: We'd have to pass by value, and not reference, because we'd be querying the contents of the deque on the other side of a `co_await`.
* Pass a `winrt...IVector`: This is possible, again passing by value, but the copy would only increment a refcount to the underlying winrt object. So that's an option.
* _not_ name this member `_startupActions`, and instead just name it `_actionQueue` or something.
* Stash a second member variable `std::deque<...> _actionQueue` in the TerminalPage, to store the non startup actions. the `initial` param would control which deque we use. 

I think option 2 is probably the cleanest here